pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -250,28 +297,49 @@ def cluster_environment(self) -> ClusterEnvironment:
         return self._cluster_environment
 
     @property
-    def on_cpu(self) -> bool:
-        return self._device_type == DeviceType.CPU
+    def has_cpu(self) -> bool:
+        return True
+
+    @property
+    def use_cpu(self) -> bool:
+        return self._accelerator_type == DeviceType.CPU
+
+    @property
+    def has_gpu(self) -> bool:
```

Comment:
based on the description below in the comment, this name is misleading IMO.

My suggestion:
```
has_gpu -> check if gpu is available on the machine
want_gpu -> if the user requests GPU
use_gpu -> if Lightning will actually choose it
```

I think that if these propertes are not 100% undoubtedly doing what their name suggests I think it will again be problematic to debug this accelerator connector as we simply cannot trust the selection process to do what the code language suggests it is doing.

maybe I'm over-worried for no reason

Reply:
I believe we could derive `want_gpu` from `has_gpu` and `use_gpu`, and it is something that hasn't been required yet.

Reply:
You mean derive `use_gpu` from `want_gpu` and `has_gpu`. The other way around does not make sense to me. 

> and it is something that hasn't been required yet.

not according to your comments in the code, where you specifically write:

> Here, we are not checking for GPU availability, but instead if User has passed
>         # `gpus` to Trainer for training.

Reply:
@awaelchli I am looking at it a different way. I am saying `has_gpu` means we have GPUs to play with, based on User input. 

```
want_gpu -> if the user requests GPU
use_gpu -> if Lightning will actually choose it
```

Isn't `want_gpu` and `use_gpu` the same thing? 

Iterating again.

```
has_gpu -> Answers if we have GPUs to work on (based on User input, not GPUs availability)
use_gpu -> Will Lightning use it? Based on `accelerator` and `has_gpu`
```

Your proposal
```
has_gpu -> check if gpu is available on the machine
want_gpu -> if the user requests GPU
```
We wouldn't really ad extra `want_gpu`, because we parse the User's `gpus` input. It will throw an error if the GPUs are not available.